includes/subscriptions.php

```diff
@@ -234,10 +234,28 @@ function delete_subscriptions( $post_id ) {
  * @since  1.0
  */
 function send_notifications( $post_id ) {
-	if ( ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) || wp_is_post_revision( $post_id ) || ! current_user_can( 'edit_post', $post_id ) ) {
+	if ( ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) || wp_is_post_revision( $post_id ) ) {
 		return;
 	}
 
+	if ( ! wp_doing_cron() ) { //phpcs:ignore
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+
+		/**
+		 * Add possibility to send notification in background
+		 *
+		 * @param bool  true    Whether run 'send notification' in background or not, default 'false'
+		 * @param array $params request data
+		 */
+		$allow_send_notification = apply_filters( 'dt_allow_send_notifications', true, $post_id );
```

Comment:
needs a description and better naming for the filter.

Reply:
Well, we can change it to `dt_allow_subscription_update`, which is more descriptive perhaps.

By using this filter we can terminate function redistribution process and run it in background. It can be helpful in cases when resource consumption is high for that process. Here is a simple example how it can be implemented:

    add_filter( 'dt_allow_subscription_update', 'schedule_send_notifications', 10, 2 );
    
    function schedule_send_notifications( $allow_send_notifications, $post_id ) {
	if ( ! wp_next_scheduled( 'dt_redistribute_posts_hook' ) ) {
		wp_schedule_single_event( time(), 'dt_redistribute_posts_hook', [ $post_id ] );
	}

	return false;
    }

Here we are scheduling an event to run in background, and by returning `false` terminating post redistribution function execution. Then we can hook the post redistribution function to scheduled action or write a custom function to redistribute post.

    add_action( 'dt_redistribute_posts_hook', 'redistribute_post', 10, 1 );
    
    function redistribute_posts( $post_id ) {
	 \Distributor\Subscriptions\send_notifications( $post_id );
    }